networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -54,180 +49,145 @@ impl <T : SuzukaExecutor + Send + Sync + Clone>SuzukaPartialNode<T> {
+					let serialized_transaction = serde_json::to_vec(&transaction)?;
+					transactions.push(BlobWrite { data: serialized_transaction });
+				},
+				Err(_) => {
+					break;
+				},
+			}
+
+			if std::time::Instant::now() > end_time {
+				break;
+			}
+		}
+
+		if transactions.len() > 0 {
+			let client_ptr = self.light_node_client.clone();
+			let mut light_node_client = client_ptr.write().await;
+			light_node_client.batch_write(BatchWriteRequest { blobs: transactions }).await?;
+			println!("Wrote transactions to DA");
+		}
+
+		Ok(())
+	}
 
-            self.executor.run_background_tasks().await?;
+	pub async fn write_transactions_to_da(&self) -> Result<(), anyhow::Error> {
+		loop {
+			self.tick_write_transactions_to_da().await?;
+		}
+	}
 
-            Ok(())
+	// receive transactions from the transaction channel and send them to be executed
+	// ! This assumes the m1 da light node is running sequencer mode
+	pub async fn read_blocks_from_da(&self) -> Result<(), anyhow::Error> {
+		let block_head_height = self.executor.get_block_head_height().await?;
+
+		let mut stream = {
+			let client_ptr = self.light_node_client.clone();
+			let mut light_node_client = client_ptr.write().await;
+			light_node_client
+				.stream_read_from_height(StreamReadFromHeightRequest { height: block_head_height })
+				.await?
+		}
+		.into_inner();
+
+		while let Some(blob) = stream.next().await {
+			println!("Stream hot!");
+			// get the block
+			let block_bytes = match blob?
+				.blob
+				.ok_or(anyhow::anyhow!("No blob in response"))?
+				.blob_type
+				.ok_or(anyhow::anyhow!("No blob type in response"))?
+			{
+				blob_response::BlobType::SequencedBlobBlock(blob) => blob.data,
+				_ => {
+					anyhow::bail!("Invalid blob type in response")
+				},
+			};
+
+			// get the block
+			let block: Block = serde_json::from_slice(&block_bytes)?;
+			println!("Received block: {:?}", block);
+
+			// get the transactions
+			let mut block_transactions = Vec::new();
+			for transaction in block.transactions {
+				let signed_transaction: SignedTransaction = serde_json::from_slice(&transaction.0)?;
+				let signature_verified_transaction = SignatureVerifiedTransaction::Valid(
+					Transaction::UserTransaction(signed_transaction),
+				);
+				block_transactions.push(signature_verified_transaction);
+			}
+
+			// form the executable transactions vec
+			let block = ExecutableTransactions::Unsharded(block_transactions);
+
+			// hash the block bytes
+			let mut hasher = sha2::Sha256::new();
+			hasher.update(&block_bytes);
+			let slice = hasher.finalize();
+			let block_hash = HashValue::from_slice(slice.as_slice())?;
+
+			// form the executable block and execute it
+			let executable_block = ExecutableBlock::new(block_hash, block);
+			let block_id = executable_block.block_id;
+			self.executor.execute_block(FinalityMode::Opt, executable_block).await?;
+
+			println!("Executed block: {:?}", block_id);
```

Comment:
Again

Reply:
These are resolved in a previous PR, will merge instead of change.